applications/MappingApplication/custom_searching/interface_communicator_mpi.cpp

```diff
@@ -0,0 +1,214 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//					 Kratos default license: kratos/license.txt
+//
+//  Main authors:    Philipp Bucher, Jordi Cotela
+//
+// See Master-Thesis P.Bucher
+// "Development and Implementation of a Parallel
+//  Framework for Non-Matching Grid Mapping"
+
+// System includes
+
+// External includes
+#include "mpi.h"
+
+// Project includes
+#include "interface_communicator_mpi.h"
+#include "custom_utilities/mapper_utilities.h"
+
+namespace Kratos
+{
+
+typedef std::size_t SizeType;
+
+/***********************************************************************************/
+/* PUBLIC Methods */
+/***********************************************************************************/
+InterfaceCommunicatorMPI::InterfaceCommunicatorMPI(ModelPart& rModelPartOrigin,
```

Comment:
@RiccardoRossi @roigcarlo this file contains the data-exchange in MPI
@RiccardoRossi was asking me abt this some time ago in relation to the global-pointers
I think I am doing sth similar (serializing objs and sending them around)

I.e. maybe this is of help for you

@roigcarlo FYI th main functionalities of this file were taken from the MPI-communicator, `AsyncSendAndReceiveObjects` and reimplemented with more recent features